bool front_detected
bool left_detected
bool right_detected